evalml/objectives/objective_base.py

```diff
@@ -10,6 +10,8 @@
 class ObjectiveBase(ABC):
     """Base class for all objectives."""
 
+    problem_types = None
```

Comment:
To allow ObjectiveBase to fall under the same `objective_base_class.rst` template as the other base classes